plugins/woocommerce/includes/abstracts/abstract-wc-payment-gateway.php

```diff
@@ -394,13 +393,23 @@ public function validate_fields() {
 	}
 
 	/**
-	 * If There are no payment fields show the description if set.
-	 * Override this in your gateway if you have some.
+	 * Default payment fields display. Override this in your gateway to customize displayed fields.
+	 *
+	 * By default this renders the payment gateway description.
+	 *
+	 * @since 1.5.7
 	 */
 	public function payment_fields() {
-		$description = $this->get_description();
-		if ( $description ) {
-			echo wpautop( wptexturize( $description ) ); // @codingStandardsIgnoreLine.
+		$description              = $this->get_description();
+		$has_extended_description = $description !== $this->description;
```

Comment:
Sorry about wondering, but I am wondering 😅
What if previous saved data contained iframes (possibly malicious) and this is now masked by some 3rd party. Disabling that code would re-enable the saved description, that would require an edit to be stripped out of malicious code. It is farfetched, I know, but maybe we could wp_kses_post() the attribute too 🤔
Thoughts?

Reply:
There is no change in logic if we cannot determine the contents of `get_description`. Kses will only apply if the description value + getter is left as default. This avoids conflicts with plugins that extended the class.